src/main/scala/java/security/SecureRandom.scala

```diff
@@ -0,0 +1,92 @@
+/*
+ * scalajs-fake-insecure-java-securerandom (https://github.com/scala-js/scala-js-fake-insecure-java-securerandom)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package java.security
+
+import scala.scalajs.js
+import scala.scalajs.js.typedarray._
+
+// The seed in java.util.Random will be unused, so set to 0L instead of having to generate one
+class SecureRandom() extends java.util.Random(0L) {
+  // Make sure to resolve the appropriate function no later than the first instantiation
+  private val getRandomValuesFun = SecureRandom.getRandomValuesFun
+
+  // setSeed has no effect
+  override def setSeed(x: Long): Unit = ()
+
+  override def nextBytes(bytes: Array[Byte]): Unit = {
+    val len = bytes.length
+    val buffer = new Int8Array(len)
+    getRandomValuesFun(buffer)
+    var i = 0
+    while (i != len) {
+      bytes(i) = buffer(i)
+      i += 1
+    }
+  }
+
+  override protected final def next(numBits: Int): Int = {
+    if (numBits == 0) {
```

Comment:
While you're here, protect against negative `numBits` as well? (OTOH it's protected, so maybe OK).